### ISSUE TITLE
App crashes on mac

### ISSUE THREAD

**Issue by coke** [git-learning-game/oh-my-git#60]
``` $ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.15.7
BuildVersion:	19H114
```

Run App

Click "Levels"
Click "Living Dangerously"

what looks like a level pops up... and then the app closes.

If I run the app from the command line, I get the same behavior and this trace:

```
$ /Applications/Oh\ My\ Git\!.app/Contents/MacOS/Oh\ My\ Git\!
arguments
0: /Applications/Oh My Git!.app/Contents/MacOS/Oh My Git!
Godot Engine v3.2.3.stable.official - https://godotengine.org
OpenGL ES 3.0 Renderer: Intel Iris OpenGL Engine

ERROR: call: Refusing to delete directory /Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/repos/yours/ that does not start with /Users/coke/Library/Application Support/Oh My Git/tmp/
   At: modules/gdscript/gdscript_functions.cpp:803.
FATAL ERROR: Refusing to delete directory /Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/repos/yours/ that does not start with /Users/coke/Library/Application Support/Oh My Git/tmp/
ERROR: call: OS.execute failed: bash [-c, "'export HOME='"'"'/Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/'"'"';export PATH='"'"'/Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/:'"'"'"$PATH";cd '"'"'/Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/'"'"' || exit 1;mkdir '"'"'/Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/repos/yours/'"'"''"] Output: mkdir: /Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/repos/yours/: File exists

Exit Code 1
   At: modules/gdscript/gdscript_functions.cpp:803.
FATAL ERROR: OS.execute failed: bash [-c, "'export HOME='"'"'/Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/'"'"';export PATH='"'"'/Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/:'"'"'"$PATH";cd '"'"'/Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/'"'"' || exit 1;mkdir '"'"'/Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/repos/yours/'"'"''"] Output: mkdir: /Volumes/BUKKIT/Users/coke/Library/Application Support/Oh My Git/tmp/repos/yours/: File exists

Exit Code 1
WARNING: cleanup: ObjectDB instances leaked at exit (run with --verbose for details).
   At: core/object.cpp:2135.
ERROR: clear: Resources still in use at exit (run with --verbose for details).
   At: core/resource.cpp:477.
ERROR: cleanup: Condition "allocs_used > 0" is true.
   At: core/pool_vector.cpp:69.
```

**Comment by EckisWelt** [comment]
Me too

**Comment by blinry** [comment]
Thanks for reporting! :)

When we delete stuff, we are extra careful not to delete the wrong stuff. In your case, because the /Users directory is on an external hard drive (I guess?), this check fails. The function `careful_delete()` in `scenes/helpers.gd` needs to be extended to handle cases like that, as well.

**Comment by coke** [comment]
Yes, home dir is on an external drive.

**Comment by coke** [comment]
Instead of building the home directory based on user id, I recommend using the HOME variable.

**Comment by linefeed** [comment]
Same issue here on Windows, where the APPDATA-directory has been relocated to a different drive.

<pre>
ERROR: call: Refusing to delete directory d:/[USERNAME]/appdata/roaming/oh my git/tmp/repos/yours/ that does not start with c:/users/[USERNAME]/appdata/roaming/oh my git/tmp/
</pre>

ENV looks like this:

HOMEDRIVE=C:
HOMEPATH=\Users\[USERNAME]
APPDATA=D:\[USERNAME]\AppData\Roaming
USERPROFILE=C:\Users\[USERNAME]

Working dir is using APPDATA location, which is fine, but safety check uses win default.

**Comment by blinry** [comment]
This fix should help, please verify as soon as 0.6.4 is published later today! :)
